qiita_db/test/test_metadata_template.py

```diff
@@ -652,6 +775,133 @@ def test_create(self):
                ['2.Sample3', "Value for sample 3"]]
         self.assertEqual(obs, exp)
 
+    def test_create_int_prefix(self):
+        """Creates a new SampleTemplate"""
+        st = SampleTemplate.create(self.metadata_int_pref, self.new_study)
+        # The returned object has the correct id
+        self.assertEqual(st.id, 2)
+
+        # The relevant rows to required_sample_info have been added.
+        obs = self.conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.required_sample_info WHERE study_id=2")
+        # sample_id study_id physical_location has_physical_specimen
+        # has_extracted_data sample_type required_sample_info_status_id
+        # collection_timestamp host_subject_id description
+        exp = [["2.12.Sample1", 2, "location1", True, True, "type1", 1,
```

Comment:
That's not correct. If the sample has been added to the database (i.e. it
appears on required_sample_info table) then it is the study id, because all
the samples are always prefixed.

If we think that another character is need, I'm fine with modifying it, but
the only characters that are accepted in the sample ids in QIIME are
letters, numbers and '.' so I don't know what else we can use...

Also, having the study id in the sample name is a convenience, so we can
ensure unique sample ids across the entire database; but it is meaningless
in any other situation...

2015-02-11 13:56 GMT-08:00 Daniel McDonald <notifications@github.com>:

> In qiita_db/test/test_metadata_template.py
> <https://github.com/biocore/qiita/pull/843#discussion_r24537536>:
>
> > @@ -652,6 +775,133 @@ def test_create(self):
> >                 ['2.Sample3', "Value for sample 3"]]
> >          self.assertEqual(obs, exp)
> >
> > +    def test_create_int_prefix(self):
> > +        """Creates a new SampleTemplate"""
> > +        st = SampleTemplate.create(self.metadata_int_pref, self.new_study)
> > +        # The returned object has the correct id
> > +        self.assertEqual(st.id, 2)
> > +
> > +        # The relevant rows to required_sample_info have been added.
> > +        obs = self.conn_handler.execute_fetchall(
> > +            "SELECT * FROM qiita.required_sample_info WHERE study_id=2")
> > +        # sample_id study_id physical_location has_physical_specimen
> > +        # has_extracted_data sample_type required_sample_info_status_id
> > +        # collection_timestamp host_subject_id description
> > +        exp = [["2.12.Sample1", 2, "location1", True, True, "type1", 1,
>
> but even within qiita, we don't actually know unless we know exactly where
> the value came from.
> … <#14b7aa4b1aaafbe5_>
> On Wed, Feb 11, 2015 at 2:54 PM, josenavas <notifications@github.com>
> wrote: In qiita_db/test/test_metadata_template.py <
> https://github.com/biocore/qiita/pull/843#discussion_r24537379>: > @@
> -652,6 +775,133 @@ def test_create(self): > ['2.Sample3', "Value for sample
> 3"]] > self.assertEqual(obs, exp) > > + def test_create_int_prefix(self): >
> + """Creates a new SampleTemplate""" > + st =
> SampleTemplate.create(self.metadata_int_pref, self.new_study) > + # The
> returned object has the correct id > + self.assertEqual(st.id, 2) > + > +
> # The relevant rows to required_sample_info have been added. > + obs =
> self.conn_handler.execute_fetchall( > + "SELECT * FROM
> qiita.required_sample_info WHERE study_id=2") > + # sample_id study_id
> physical_location has_physical_specimen > + # has_extracted_data
> sample_type required_sample_info_status_id > + # collection_timestamp
> host_subject_id description > + exp = [["2.12.Sample1", 2, "location1",
> True, True, "type1", 1, Hahaha I know that was your next question. And I
> agree that without context, that is really confusing. If that string is a
> sample id on Qiita, yes it does. If it is outside Qiita, the study id is
> meaningless so you can arbitrarily decide if it does or not... 2015-02-11
> 13:50 GMT-08:00 Daniel McDonald <notifications@github.com>: …
> <#14b7aa3182127623_> In qiita_db/test/test_metadata_template.py <
> https://github.com/biocore/qiita/pull/843#discussion_r24537015>: > @@
> -652,6 +775,133 @@ def test_create(self): > ['2.Sample3', "Value for sample
> 3"]] > self.assertEqual(obs, exp) > > + def test_create_int_prefix(self): >
> + """Creates a new SampleTemplate""" > + st =
> SampleTemplate.create(self.metadata_int_pref, self.new_study) > + # The
> returned object has the correct id > + self.assertEqual(st.id, 2) > + > +
> # The relevant rows to required_sample_info have been added. > + obs =
> self.conn_handler.execute_fetchall( > + "SELECT * FROM
> qiita.required_sample_info WHERE study_id=2") > + # sample_id study_id
> physical_location has_physical_specimen > + # has_extracted_data
> sample_type required_sample_info_status_id > + # collection_timestamp
> host_subject_id description > + exp = [["2.12.Sample1", 2, "location1",
> True, True, "type1", 1, Okay, let me rephrase, does "123.foo" contain a
> study ID prefix or not? — Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/843/files#r24537015>. -- Jose
> Navas — Reply to this email directly or view it on GitHub <
> https://github.com/biocore/qiita/pull/843/files#r24537379>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/843/files#r24537536>.
>



-- 
Jose Navas